[Feature Addition]: Web App for Vehicle Live Risk Prediction

shivansh-2003 opened the pull request:
# Pull Request for ML-Crate 💡


## Issue Title:[Feature Addition]: Web App for Vehicle Live Risk Prediction #602

- **Info about the related issue (Aim of the project)** : 'The aim is to create a web app for the Vehicle Live Risk Prediction project.
- **Name:** Shivansh Mahajan
- **GitHub ID:** https://github.com/shivansh-2003
- **Idenitfy yourself:** Data Scientist


<!-- Mention the following details and these are mandatory -->

Closes: #issue number that will be closed through this PR

### Describe the add-ons or changes you've made 📃
1. I have updated xgboost.py file 
2. I have created Web app directory 
3. I have created app.py and readme.md
4. I have also created pickle file in model folder and Readme.md file
5. I have updated README.md 
6. I also created REadme.md file in dataset folder

Give a clear description of what have you added or modifications made

## Type of change ☑️

What sort of change have you made:
<!--
Example how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [☑️] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

Describe how it has been tested
I have tested the vehicle live risk prediction app by running the app on Streamlit and verifying the result by testing various inputs .




## Checklist: ☑️
<!--
Example how to mark a checkbox:-
- [☑️] My code follows the code style of this project.
-->
- [☑️] My code follows the guidelines of this project.
- [☑️ ] I have performed a self-review of my own code.
- [☑️] I have commented my code, particularly wherever it was hard to understand.
- [☑️] I have made corresponding changes to the documentation.
- [☑️] My changes generate no new warnings.
- [☑️] I have added things that prove my fix is effective or that my feature works.
- [☑️] Any dependent changes have been merged and published in downstream modules.

abhisheks008 commented:
Create a single file for the models implemented, do not use so many files for every models you have implemented.
@shivansh-2003

shivansh-2003 commented:
sir i have deleted the extra models file and only kept xgboost file and if any other changes required please let me know

abhisheks008 reviewed (commented):
Have you added a new jupyter notebook file and did the model creation?
@shivansh-2003

shivansh-2003 commented:
I have modified the existing xgboost file with visualization and madel training and pickle file creation

abhisheks008 reviewed (changes requested):
1. Remove the `.DS_Store` files.
2. The file name is `README.md`. Please update the file name everywhere in the project.
3. The folder name should be `Web App`. Also upload the demonstration video file inside the Web App folder.
4. Why have you removed the `.ipynb` files. You can keep them, I don't that will effect your contributions.

shivansh-2003 commented:
All the requested changes have been catered . Video demonstration is uploaded in web app folder and link is also provided in readme file . I have restored all the extra model files that i have deleted and renamed all readme file . Deleted .ds/store file.

abhisheks008 commented on Vehicle Live Risk Prediction/Dataset/REadme.md:
```diff
@@ -0,0 +1,31 @@
+# Live Vehicle Risk Prediction Dataset
+
+## Overview
```
rename this README file

abhisheks008 commented on Vehicle Live Risk Prediction/Model/Readme.md:
```diff
@@ -0,0 +1,94 @@
+# Live Vehicle Risk Detection using XGBoost Model
+
```
rename this file as README.md

abhisheks008 commented on Vehicle Live Risk Prediction/Web App/README.md:
```diff
@@ -0,0 +1,55 @@
+   - Provides information about the model explanation and purpose of the application.
+
+### Video Demonstration 🎥
+[Watch the video](https://vimeo.com/953208773?share=copy)
```
Instead of hyperlinking the demo video in the README, copy and paste the same video to the README from your local system directly, it'll work better as we can get an embedded version of the video.

shivansh-2003 replied:
sir i am trying to embed video from my local system but it does not work by copying directly

abhisheks008 reviewed (changes requested):
Also remove the `.DS_Store` file outside the project folder.

shivansh-2003 commented:
I have completed all issues except of video embed feature because a/q to your instruction embed is not created by copying local file video

abhisheks008 commented:
> I have completed all issues except of video embed feature because a/q to your instruction embed is not created by copying local file video

Just drag and drop the video file from your local system to the README while editing it.

shivansh-2003 commented:
> Just drag and drop the video file from your local system to the README while editing it.
Done sir

shivansh-2003 commented:
sir i have completed all the requested task

abhisheks008 reviewed (approved):
Approved @shivansh-2003

abhisheks008 commented:
Please share your email address for further communication.
@shivansh-2003

abhisheks008 commented:
> Please share your email address for further communication. @shivansh-2003

Share it ASAP.

shivansh-2003 commented:
shivansh.m2003@gmail.com
